src/main/java/org/example/jackson/bench/NumberUtil.java

```diff
@@ -66,4 +68,108 @@ public static long alternateParseLong(String s) {
+            offset = 1;
+        } else {
+            if (len > 19) {
+                return Long.parseLong(s);
+            }
+            offset = 0;
+        }
+        long num = 0L;
+        for (int i = offset; i < len; i++) {
+            char c = s.charAt(i);
+            if (c > '9' || c < '0') {
+                return Long.parseLong(s);
+            }
+            num = (num * 10) + (c - '0');
+        }
+        return neg ? -num : num;
+    }
+    
+    public static long alternateParseSwitch(String s) {
+        /* Ok: let's keep strategy simple: ignoring optional minus sign,
+         * we'll accept 1 - 19 digits and parse things efficiently;
+         * otherwise just defer to JDK parse functionality.
+         */
+        int len = s.length();
+        boolean neg = s.charAt(0) == '-';
+        int off;
+        // must have 1 - 19 digits after optional sign:
+        // negative?
+        if (neg) {
+            if (len == 1 || len > 20) {
+                return Long.parseLong(s);
+            }
+            off = 1;
+        } else {
+            if (len > 19) {
+                return Long.parseLong(s);
+            }
+            off = 0;
+        }
+        long num = s.charAt(len - 1) - '0';
+        
+        switch(len- off) {
+            case 19: 
+                num += (digitAt(s, len - 19)) * 1000000000000000000L;
+            case 18: 
+                num += (digitAt(s, len - 18)) * 100000000000000000L;
+            case 17: 
+                num += (digitAt(s, len - 17)) * 10000000000000000L;
+            case 16: 
+                num += (digitAt(s, len - 16)) * 1000000000000000L;
+            case 15: 
+                num += (digitAt(s, len - 15)) * 100000000000000L;
+            case 14: 
+                num += (digitAt(s, len - 14)) * 10000000000000L;
+            case 13: 
+                num += (digitAt(s, len - 13)) * 1000000000000L;
+            case 12: 
+                num += (digitAt(s, len - 12)) * 100000000000L;
+            case 11: 
+                num += (digitAt(s, len - 11)) * 10000000000L;
+            case 10: 
+                num += (digitAt(s, len - 10)) * 1000000000L;
+            case 9: 
+              num += (digitAt(s, len - 9)) * 100000000L;
+            case 8: 
+              num += (digitAt(s, len - 8)) * 10000000L;
+            case 7: 
+              num += (digitAt(s, len - 7)) * 1000000L;
+            case 6: 
+              num += (digitAt(s, len - 6)) * 100000L;
+            case 5: 
+              num += (digitAt(s, len - 5)) * 10000L;
+            case 4: 
+              num += (digitAt(s, len - 4)) * 1000L;
+            case 3: 
+              num += (digitAt(s, len - 3)) * 100L;
+            case 2: 
+              num += (digitAt(s, len - 2)) * 10L;
+        }
+        return neg ? -num : num;
+    }
+    
+    private static long digitAt(String s, int i) {
+        char c = s.charAt(i);
+        if (c > '9' || c < '0') {
+            return Long.parseLong(s);
```

Comment:
in hindsight, this is wrong - when the Long.parseLong(s) is returned, there is no indicator that we parsed the whole string as opposed to one char - so we will keep iterating through the chars